### PR TITLE
[WIP] RSC can see list of Staffing Companies

### PR DESCRIPTION
**Pull request by RhonGabriel**
# PT 
https://www.pivotaltracker.com/story/show/161660051

### Included in PR

* Adding staffing companies model 
* Adding staffing companies  controller 
* Adding staffing companies  index view  
* Adding feature and step definition for RSC can see list of staffing companies 
* Adding factory bot for staffing companies 
 Adding styling (and extracting tailwind components)
screenshot 2018-11-02 at 11 57 13

### CODE REVIEW

**Review by Noelryn — APPROVED** [review]
Looks good to me

File: app/models/staffco.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class Staffco < ApplicationRecord
```
Comment (rupertlion) [inline app/models/staffco.rb:1]: we need to validate columns here

File: app/views/jobs/index.html.haml [hunk 2]
```diff
@@ -19,4 +19,17 @@
         %td
           %button -
   #shifts
-   
\ No newline at end of file
+
+.card
```
Comment (rupertlion) [inline app/views/jobs/index.html.haml]: this code should all be in staffcos/index (I think)

File: app/views/staffcos/index.html.haml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+%h1 Staffcos#index
```
Comment (rupertlion) [inline app/views/staffcos/index.html.haml]: This file should have all the staffco view/index/list items

File: config/routes.rb [hunk 4]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  get 'staffcos/index'
```
Comment (rupertlion) [inline config/routes.rb]: Old way of doing - add as a resource

File: app/views/jobs/index.html.haml [hunk 5]
```diff
@@ -1,22 +1,18 @@
...
+      %td
+        %button -
+    %tr
+      %td Job 2
+      %td Posted
+      %td
+        %button Edit
+      %td
+        %button -
+#shifts
```
Comment (rupertlion) [inline app/views/jobs/index.html.haml]: why do we need this here? I wonder whether it is required at all now

File: app/views/partials/_sidebar.html.haml [hunk 6]
```diff
@@ -1,3 +1,5 @@
-%button.zdh-button= t :jobs
-%button.zdh-button= t :hospitals
-%button.zdh-button= t :staffcos
\ No newline at end of file
+.grid-container
+  #sidebar
+    %button.zdh-button= t :jobs
+    %button.zdh-button= t :hospitals
+    %button.zdh-button= t :staffcos
```
Comment (rupertlion) [inline app/views/partials/_sidebar.html.haml]: why don't you link this to the staffco index using =link_to?

File: config/routes.rb [hunk 7]
```diff
@@ -1,5 +1,7 @@
...
```
Comment (rupertlion) [inline config/routes.rb]: still need to remove this line as we have the resource below

File: app/views/partials/_sidebar.html.haml (hunk 6 above)
Comment (RhonGabriel) [reply app/views/partials/_sidebar.html.haml]: done

File: config/routes.rb (hunk 7 above)
Comment (RhonGabriel) [reply config/routes.rb]: solved

**Review by Prozak8 — APPROVED** [review]

**Pull request merged**